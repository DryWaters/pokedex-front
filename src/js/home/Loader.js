import React from 'react';

import LoadingGif from '../../assets/loading_icon.gif';

class Loader extends React.Component {
  render () {
    return (
      <div align="center">
        <img key="loader" src={LoadingGif} alt="Loading more Pokemon" />
      </div>
    );
  }
}

export default Loader;
