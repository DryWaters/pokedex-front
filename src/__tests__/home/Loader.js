import React from 'react';
import { shallow } from 'enzyme';

import Loader from '../../js/home/Loader';

describe('Loader Component', () => {
  it('should render Loader component without crashing', () => {
    shallow(<Loader />);
  });

  it('should match Loader component snapshot', () => {
    const wrapper = shallow(<Loader />);
    expect(wrapper).toMatchSnapshot();
  });
});
