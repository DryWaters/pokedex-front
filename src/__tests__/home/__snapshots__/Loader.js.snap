// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component should match Loader component snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "align": "center",
      "children": <img
        alt="Loading more Pokémon"
        src="loading_icon.gif"
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": "loader",
      "nodeType": "host",
      "props": Object {
        "alt": "Loading more Pokémon",
        "src": "loading_icon.gif",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "align": "center",
        "children": <img
          alt="Loading more Pokémon"
          src="loading_icon.gif"
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": "loader",
        "nodeType": "host",
        "props": Object {
          "alt": "Loading more Pokémon",
          "src": "loading_icon.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
